base/src/org/compiere/model/Tax.java

```diff
@@ -191,70 +197,76 @@ public static int getCharge (Properties ctx, int chargeId,
 			 + " AND sl.C_BPartner_Location_ID=?";
 		PreparedStatement pstmt = null;
 		ResultSet rs = null;
-		try
-		{
-			pstmt = DB.prepareStatement (sql, trxName);
-			pstmt.setInt (1, warehouseId);
-			pstmt.setInt (2, chargeId);
-			pstmt.setInt (3, orgId);
-			pstmt.setInt (4, billPartnerLocationId);
-			pstmt.setInt (5, shipPartnerLocationId);
-			rs = pstmt.executeQuery ();
-			boolean found = false;
-			if (rs.next ())
-			{
-				taxCategoryId = rs.getInt (1);
-				billFromLocationId = rs.getInt (2);
-				billToLocationId = rs.getInt (3);
-				isSOTaxExempt = rs.getString (4);
-				isPOTaxExempt = rs.getString (5);
-				isTaxExempt = IsSOTrx ? isSOTaxExempt : isPOTaxExempt;
-				shipFromLocationId = rs.getInt (6);
-				shipToLocationId = rs.getInt (7);
-				found = true;
-			}
-			DB.close(rs, pstmt);
-			//
-			if (!found)
-			{
+
+		List params = List.of(warehouseId, chargeId, orgId, billPartnerLocationId, shipPartnerLocationId);
+		DB.runResultSetFunction.apply(trxName, sql, params, resultSet -> {
+			List<Tuple7<Integer, Integer, Integer, String, String, Integer, Integer>> taxes = new ResultSetIterable<>(resultSet, row -> Tuple.of(
+					row.getInt(1) ,
+					row.getInt(2) ,
+					row.getInt(3) ,
+					row.getString(4),
+					row.getString(5),
+					row.getInt(6) ,
+					row.getInt(7)
+			)).toList();

```

Comment:
@nsarlabos To increase the readability in the code it is better to use the names of the columns instead of the position of what the result set returns, in this way you save the one who is understanding the code exactly which column corresponds to which thing, for On the other hand, using numbers is prone to errors.

ie change row.getInt(1) to  row.getInt("C_TaxCategory_ID");

Reply:
Hi @e-Evolution It's solved